bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -1025,24 +1047,32 @@ class MastodonClient private constructor(
          * connection are _not_ caught by this library.
          */
         fun build(): MastodonClient {
+            val httpClient = okHttpClientBuilder
+                .addNetworkInterceptor(AuthorizationInterceptor(accessToken))
+                .readTimeout(readTimeoutSeconds, TimeUnit.SECONDS)
+                .writeTimeout(writeTimeoutSeconds, TimeUnit.SECONDS)
+                .connectTimeout(connectTimeoutSeconds, TimeUnit.SECONDS)
+                .build()
+
+            // Get instance version first as this may fail if the server doesn't appear to run Mastodon
+            val instanceVersion = getInstanceVersion()
+
+            val instance = getInstance(httpClient)
+            val streamingUrl = getStreamingApiUrl(
+                instance = instance,
+                fallbackUrl = HttpUrl.Builder().scheme(scheme).host(instanceName)::toString
+            )
+
             return MastodonClient(
                 instanceName = instanceName,
-                client = okHttpClientBuilder
-                    .addNetworkInterceptor(AuthorizationInterceptor(accessToken))
-                    .readTimeout(readTimeoutSeconds, TimeUnit.SECONDS)
-                    .writeTimeout(writeTimeoutSeconds, TimeUnit.SECONDS)
-                    .connectTimeout(connectTimeoutSeconds, TimeUnit.SECONDS)
-                    .build(),
+                instance = instance,
+                client = httpClient,
                 accessToken = accessToken,
                 debug = debug,
-                instanceVersion = getInstanceVersion(),
+                instanceVersion = instanceVersion,
                 scheme = scheme,
                 port = port,
-                streamingUrl = getStreamingApiUrl(
-                    fallbackUrl = {
-                        HttpUrl.Builder().scheme(scheme).host(instanceName).toString()
-                    }
-                )
+                streamingUrl = streamingUrl
             )
         }
     }
```

Comment:
With `instance` available, this now contains duplicate information which we could get rid of and by doing so remove some parameters from the `MastodonClient` constructor. This constructor is private, so no breaking change. Specifically:
- `instanceName` is also available as `instance.domain`
- `instanceVersion` is also available as `instance.version`
- `streamingUrl` is also available as `instance.configuration.urls.streaming` - in this case, if we still want to rely on a fallback URL, the whole process of getting either the configured or a fallback URL could probably be moved into the client itself.

Reply:
I’ve taken both into account in a563f5e 😊